packages/abi-typegen/src/types/GenericType.ts

```diff
@@ -0,0 +1,27 @@
+import type { IType } from '../interfaces/IType';
+
+import { AType } from './AType';
+
+export class GenericType extends AType implements IType {
+  public static swayType = 'generic T';
+
+  public name = 'generic';
+
+  static MATCH_REGEX: RegExp = /^generic ([^\s]+)$/m;
+
+  static isSuitableFor(params: { type: string }) {
+    return GenericType.MATCH_REGEX.test(params.type);
+  }
+
+  public parseComponentsAttributes(_params: { types: IType[] }) {
+    const matched = this.rawAbiType.type.match(GenericType.MATCH_REGEX);
+    const label = matched![1];
```

Comment:
It would be good to check here and throw a descriptive error if the type doesn't match. Or if is really not needed add eslint ignore comment